File: ANSv4/Catalogo4/util/IPPrivado.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace Catalogo.util
{
    class IPPrivado
    {

	    // Esta clase realiza una consulta al servidor web y obtiene    
	    // el numero de ip del servidor privado.

	    // Define como se llama este modulo para el control de errores

	    //private //const string m_sMODULENAME_ = "clsIPPrivado";
	    private CatalogoWS.Info Cliente;

	    private bool WebServiceInicializado;

	    private string m_MacAddress;
	    public bool Inicializado 
        {
		    get { return WebServiceInicializado; }
	    }

	    public string GetIP()
	    {
		    string functionReturnValue = null;

		     // ERROR: Not supported in C#: OnErrorStatement
            
		    string s = "";

		    if (!WebServiceInicializado) {
			    //Err.Raise 9999, , "Falta inicializar"
			    functionReturnValue = "";
			    return functionReturnValue;
		    }

		    s = Cliente.GetIp(m_MacAddress);
		    functionReturnValue = s.Trim();

		    return functionReturnValue;

            //AtajarError:


            //if (Err().Number == -2147467259) {
            //    Err().Clear();
            //    functionReturnValue = "";
            //    return functionReturnValue;
            //} else {
            //    Err().Raise(Err().Number, Err().Source, Err().Description);
            //}
            //return functionReturnValue;

	    }

	    public string GetIpIntranet()
	    {
		    string functionReturnValue = null;

		     // ERROR: Not supported in C#: OnErrorStatement

		    string s = "";

		    if (!WebServiceInicializado) {
			    //Err.Raise 9999, , "Falta inicializar"
			    functionReturnValue = "";
			    return functionReturnValue;
		    }

		    s = Cliente.GetIpIntranet(m_MacAddress);
            functionReturnValue = s.Trim();

		    return functionReturnValue;

            //AtajarError:


            //if (Err().Number == -2147467259) {
            //    Err().Clear();
            //    functionReturnValue = "";
            //    return functionReturnValue;
            //} else {
            //    Err().Raise(Err().Number, Err().Source, Err().Description);
            //}
            //return functionReturnValue;

	    }

        //public string GetIPCatalogo()
        //{
        //    string functionReturnValue = null;

        //     // ERROR: Not supported in C#: OnErrorStatement


        //    string s = null;

        //    if (!WebServiceInicializado) {
        //        //Err.Raise 9999, , "Falta Inicializar"
        //        functionReturnValue = "";
        //        return functionReturnValue;
        //    }

        //    s = Cliente.GetIpCatalogo(m_MacAddress);
        //    functionReturnValue = s.Trim();

        //    return functionReturnValue;

        //}

	    public IPPrivado(string ipAddress, string MacAddress)
	    {

            bool Conectado;

        VadeNuevo:
            Conectado = util.SimplePing.ping(ipAddress, 5000, 0);
            if (!Conectado)
            {
                if (MessageBox.Show("Error de Conexión al Servidor, ¿quiere intentar de nuevo?", "Atención", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.Yes)
                {
                    Global01.URL_ANS = Global01.URL_ANS2;
                    ipAddress = Global01.URL_ANS;
                    goto VadeNuevo;
                }
            }

		    if (!WebServiceInicializado) {       
                if (Conectado) {
				    Cliente = new CatalogoWS.Info();
                    //Cliente.Url = Catalogo.Properties.Settings.Default.Catalogo_CatalogoWS_Info;
                    Cliente.Url = "http://" + ipAddress + "/wsCatalogo4/Info.asmx?wsdl";
                    if (Global01.proxyServerAddress != "0.0.0.0")
                    {
					    Cliente.Proxy = new System.Net.WebProxy(Global01.proxyServerAddress);
				    }

				    WebServiceInicializado = true;
				    m_MacAddress = MacAddress;
			    } else {
				    WebServiceInicializado = false;
                    util.errorHandling.ErrorLogger.LogMessage(new Exception("IpPrivado - ws Info - No Inicializado"));
			    }
		    }

		    return;

            //.ErrorForm.show():


            //if (Err().Number == -2147024809) {
            //    // Este error se da debido
            //    WebServiceInicializado = false;
            //    Err().Clear();
            //} else {
            //    Err().Raise(Err().Number, Err().Source, Err().Description);
            //}

	    }

    }
}

File: ANSv4/CatalogoLibraryVB/IPs/IPPrivado.vb
Option Explicit On

Public Class IPPrivado
    ' Esta clase realiza una consulta al servidor web y obtiene
    ' el numero de ip del servidor privado.

    ' Define como se llama este modulo para el control de errores
    Private Const m_sMODULENAME_ As String = "clsIPPrivado"

    Private Cliente As CatalogoWS.Info
    Private WebServiceInicializado As Boolean

    Private m_MacAddress As String

    Public ReadOnly Property Inicializado() As Boolean
        Get
            Return WebServiceInicializado
        End Get
    End Property

    Public Function GetIP() As String

        On Error GoTo AtajarError

        Dim s As String

        If Not WebServiceInicializado Then
            'Err.Raise 9999, , "Falta inicializar"
            GetIP = ""
            Exit Function
        End If
        s = Cliente.GetIp(m_MacAddress)
        GetIP = Trim(s)

        Exit Function

AtajarError:
        If Err.Number = -2147467259 Then
            Err.Clear()
            GetIP = ""
            Exit Function
        Else
            Err.Raise(Err.Number, Err.Source, Err.Description)
        End If

    End Function

    Public Function GetIpIntranet() As String

        On Error GoTo AtajarError

        Dim s As String

        If Not WebServiceInicializado Then
            'Err.Raise 9999, , "Falta inicializar"
            GetIpIntranet = ""
            Exit Function
        End If

        s = Cliente.GetIpIntranet(m_MacAddress)
        GetIpIntranet = Trim(s)

        Exit Function

AtajarError:
        If Err.Number = -2147467259 Then
            Err.Clear()
            GetIpIntranet = ""
            Exit Function
        Else
            Err.Raise(Err.Number, Err.Source, Err.Description)
        End If

    End Function

    Public Function GetIPCatalogo() As String

        On Error GoTo AtajarError

        Dim s As String

        If Not WebServiceInicializado Then
            'Err.Raise 9999, , "Falta Inicializar"
            GetIPCatalogo = ""
            Exit Function
        End If

        s = Cliente.GetIpCatalogo(m_MacAddress)
        GetIPCatalogo = Trim(s)

        Exit Function

AtajarError:
        If Err.Number = -2147467259 Then
            Err.Clear()
            GetIPCatalogo = ""
            Exit Function
        Else
            Err.Raise(Err.Number, Err.Source, Err.Description)
        End If

    End Function

    Public Sub New(ByVal ipAddress As String, ByVal MacAddress As String, ByVal usaProxy As Boolean, _
                   ByVal URLProxyServer As String)
        On Error GoTo ErrorHandler

        Dim Conectado As Boolean

        Conectado = My.Computer.Network.Ping(ipAddress, 5000)
        Conectado = True
        If Not WebServiceInicializado Then
            If Conectado Then
                'Cliente = New WSCatalogo.InfoSoapClient("", "http://" & ipAddress & "/Catalogo/Info.asmx?wsdl")
                Cliente = New CatalogoWS.Info()
                ' DIEGO -> Implementar proxy!
                If usaProxy Then
                    Cliente.Proxy = New System.Net.WebProxy(URLProxyServer)
                End If

                WebServiceInicializado = True
                m_MacAddress = MacAddress
            Else
                WebServiceInicializado = False
            End If
        End If

        Exit Sub

ErrorHandler:

        If Err.Number = -2147024809 Then
            ' Este error se da debido
            WebServiceInicializado = False
            Err.Clear()
        Else
            Err.Raise(Err.Number, Err.Source, Err.Description)
        End If

    End Sub


End Class
